### ISSUE TITLE
SSL Issue

### ISSUE THREAD

**Issue by rajeshdua123** [jchristn/WatsonWebsocket#36]
Hi,

1.  We recently installed an SSL certificate on our server.  
2.  Our base socket was running without SSL before. "ws://".  Port is 9000.
3.  Everything was fine.
4.  After SSL installation, we made SSL as true in the socket program and re-ran it.
5.  The connection to the socket stopped.  I understand that this is because of certificate installation.
6.  I followed your instructions available here - https://github.com/jchristn/WatsonWebserver/wiki/Using-SSL-on-Windows
7.  When I ran netsh command to add certificate, I got a message that certificate is already added to 0.0.0.0:443.  I understand that while installing the server, it got added automatically.

8.  I tried to add this certificate to port 9000 but got an error - "SSL Certificate add failed, Error: 1312 A specified logon session does not exist. It may already have been terminated."

9.  I am sure I am missing something here.  

Please guide. 

Thanks,
Rajesh

**Comment by jchristn** [comment]
Hi @rajeshdua123 did you confirm the certificate hash assigned to 0.0.0.0:443 is the same as yours when doing ```netsh http show sslcert```?

Also when copying the certificate hash out of the MMC, paste it into notepad and then retype it on a newline.  When you execute the command ```netsh http add sslcert ipport="0.0.0.0:443" certhash="[thumbprint]" appid="{00000000-0000-0000-0000-000000000000}" certstore=My``` used the hash that you manually typed out.  There are always junk characters that you never see when pasting out of the certificate manager MMC.

Here's a link that may be helpful.  https://stackoverflow.com/questions/13076915/ssl-certificate-add-failed-when-binding-to-port

Let me know how it goes.

**Comment by rajeshdua123** [comment]
Thanks for your reply.  The instructions were great.

Here is what I did, so that it will help others as well - 
1.  We are running 4 sockets for different services. 
2.  Each socket is running on a different port. 
3.  Installed the SSL certificate on IIS (used personal instead of Web Hosting option during installation).
4.  Opened up each port on Windows Firewall.
5.  Installed the certificate for each port - used commands mentioned in your link -  https://github.com/jchristn/WatsonWebserver/wiki/Using-SSL-on-Windows
6.  In the core program, made SSL to true.  Changed the earlier "ws://{ip address}" to "wss://{ip address}".
7.  Earlier the application was using IPaddress to connect with sockets. Changed that from "ws://{ip and port}" to "wss://{domainname and port}"
8.  All is smooth and running now.

Thank you so much to write a wonderful wrapper for websockets.

Thanks and regards,
Rajesh
